packages/lib/server/checkBookingLimits.ts

```diff
@@ -1,94 +1,71 @@
-    const limitCalculations = Object.entries(parsedBookingLimits).map(
-      async ([key, limitingNumber]) =>
-        await checkBookingLimit({ key, limitingNumber, eventStartDate, eventId, returnBusyTimes })
-    );
-    await Promise.all(limitCalculations)
-      .then((res) => {
-        if (returnBusyTimes) {
-          return res;
-        }
-      })
-      .catch((error) => {
-        throw new HttpError({ message: error.message, statusCode: 401 });
-      });
-    return true;
+  if (!parsedBookingLimits) return false;
+
+  // not iterating entries to preserve types
+  const limitCalculations = ascendingLimitKeys.map((key) =>
+    checkBookingLimit({ key, limitingNumber: parsedBookingLimits[key], eventStartDate, eventId })
+  );
+
+  try {
+    return !!(await Promise.all(limitCalculations));
+  } catch (error) {
+    throw new HttpError({ message: getErrorFromUnknown(error).message, statusCode: 401 });
   }
-  return false;
 }
 
 export async function checkBookingLimit({
   eventStartDate,
   eventId,
   key,
   limitingNumber,
-  returnBusyTimes = false,
 }: {
   eventStartDate: Date;
   eventId: number;
-  key: string;
-  limitingNumber: number;
-  returnBusyTimes?: boolean;
+  key: keyof IntervalLimit;
+  limitingNumber: number | undefined;
 }) {
   {
-    const limitKey = key as keyof IntervalLimit;
-    // Take PER_DAY and turn it into day and PER_WEEK into week etc.
-    const filter = limitKey.split("_")[1].toLocaleLowerCase() as "day" | "week" | "month" | "year"; // Have to cast here
-    const startDate = dayjs(eventStartDate).startOf(filter).toDate();
-    // this is parsed above with parseBookingLimit so we know it's safe.
+    if (!limitingNumber) return;
+
+    const unit = intervalLimitKeyToUnit(key);
 
-    const endDate = dayjs(startDate).endOf(filter).toDate();
+    const startDate = dayjs(eventStartDate).startOf(unit).toDate();
+    const endDate = dayjs(eventStartDate).endOf(unit).toDate();
 
-    // This allows us to easily add it within dayjs
     const bookingsInPeriod = await prisma.booking.count({
       where: {
-        status: "ACCEPTED",
-        eventType: {
-          AND: [
-            {
-              id: eventId,
-              bookings: {
-                every: {
-                  startTime: {
-                    gte: startDate,
-                  },
-                  endTime: {
-                    lte: endDate,
-                  },
-                },
-              },
-            },
-          ],
+        status: BookingStatus.ACCEPTED,
+        eventTypeId: eventId,
+        // FIXME: bookings that overlap on one side will never be counted
+        startTime: {
+          gte: startDate,
+        },
+        endTime: {
+          lte: endDate,
         },
       },
     });
-    if (bookingsInPeriod >= limitingNumber) {
-      // This is used when getting availability
-      if (returnBusyTimes) {
-        return {
-          start: startDate,
-          end: endDate,
-        };
-      }
```

Comment:
Now unused, as we already have that data when checking availability for specific periods